tool2schema/__init__.py

```diff
@@ -10,3 +11,14 @@
     SaveGPTEnabled,
     SchemaType,
 )
+
+# Default global configuration
+CONFIG = Config()
+
+
+def _reset_config():
+    """
+    Reset the configuration to the default settings.
+    """
+    global CONFIG
+    CONFIG = Config()
```

Comment:
If this method is included here then, it shouldn't have the `_` prefix. But perhaps a better place would be within the `Config` object to facilitate an API like

```python
tool2schema.CONFIG.reset_default()
```

Reply:
I added the prefix `_` because it was meant to be an internal method used only for testing purposes. With the current configuration class it is not possible to reset the configuration because the attributes were meant to be immutable and the default values are only know in the constructor.

But here's another idea, using properties instead of class attributes:
```python
class Config:

    def __init__(self, parent: Optional[Config] = None, **settings):
        self._parent = parent
        self._settings = settings

    @property
    def ignore_parameters(self) -> list[str]:
        return self._get_setting(Config.ignore_parameters.fget.__name__, ["self", "args", "kwargs"])

    @ignore_parameters.setter
    def ignore_parameters(self, value: list[str]):
        self._set_setting(Config.ignore_parameters.fget.__name__, value)

    def reset(self):
        self._settings = {}

    def _get_setting(self, name: str, default):
        default = default if not self._parent else getattr(self._parent, name)
        return self._settings.get(name, default)

    def _set_setting(self, name: str, value):
        self._settings[name] = value
```

Advantages:
- Settings are dynamic, so if you change the parent configuration it will reflect on children
- We can document the properties
- We can have a reset method
- There are no strings with the name of the settings, thus the properties can be renamed easily

Disadvantage:
- You have to write 7 lines of code and two methods (getter/setter) to add a new setting.

Reply:
Let me know! :)

Reply:
I checked it in because I liked it better but I can always go back

Reply:
Yeah, I like it. To me the main trade-off is

We have autocomplete for the `Config` object
vs
Two methods per setting

I think that's acceptable.